### ISSUE TITLE
Seek to position sometimes does not fulfill tolerance settings

### ISSUE THREAD

**Issue by defagos** [SRGSSR/srgmediaplayer-apple#55]
For some tolerance settings and start position, and depending on where the time to seek to is located within a chunk, tolerance settings are not correctly applied, and playback might start outside the [tolerated range](https://developer.apple.com/documentation/avfoundation/avplayer/1387741-seektotime).

This issue is an `AVPlayer` issue or limitation, or there is something I did not correctly understand with `-seekToTime:toleranceBefore:toleranceAfter:completionHandler:`:

```objective-c
AVPlayerViewController *playerViewController = [[AVPlayerViewController alloc] init];
NSURL *URL = [NSURL URLWithString:@"http://stream-i.akamaihd.net/i/tj/2016/tj_20160905_full_f_965656-,101,301,701,1201,k.mp4.csmil/master.m3u8"];
AVPlayer *player = [AVPlayer playerWithURL:URL];
playerViewController.player = player;
[self presentViewController:playerViewController animated:YES completion:^{
    [player seekToTime:CMTimeMakeWithSeconds(219., NSEC_PER_SEC) toleranceBefore:CMTimeMakeWithSeconds(3., NSEC_PER_SEC) toleranceAfter:kCMTimeZero completionHandler:^(BOOL finished) {
        [player play];
    }];
}];
```

Playback would in this case be expected to start in `[216, inf]`, but will actually start at the beginning of the chunk near 212.

**Comment by defagos** [comment]
Seek is sharp for

```objective-c
[self presentViewController:playerViewController animated:YES completion:^{
    [player seekToTime:CMTimeMakeWithSeconds(219., NSEC_PER_SEC) toleranceBefore:kCMTimeZero toleranceAfter:kCMTimeZero completionHandler:^(BOOL finished) {
        [player play];
    }];
}];
```

but jumps to 212, even for small tolerances:

```objective-c
[self presentViewController:playerViewController animated:YES completion:^{
    [player seekToTime:CMTimeMakeWithSeconds(219., NSEC_PER_SEC) toleranceBefore:CMTimeMakeWithSeconds(0.1, NSEC_PER_SEC) toleranceAfter:kCMTimeZero completionHandler:^(BOOL finished) {
        [player play];
    }];
}];
```

What is puzzling me is that playback also starts at 212 if we set a small tolerance after:

```objective-c
[self presentViewController:playerViewController animated:YES completion:^{
    [player seekToTime:CMTimeMakeWithSeconds(219., NSEC_PER_SEC) toleranceBefore:kCMTimeZero toleranceAfter:CMTimeMakeWithSeconds(0.1, NSEC_PER_SEC) completionHandler:^(BOOL finished) {
        [player play];
    }];
}];
```

**Comment by defagos** [comment]
I opened a [radar](http://openradar.appspot.com/radar?id=4931655470940160) for this issue.

**Comment by defagos** [comment]
I fixed related unit tests (and ensured they break with the current behavior, highlighting the issue). This is available on the `feature/imprecise-seeks` branch.

**Comment by defagos** [comment]
Still waiting for an answer from Apple.

**Comment by defagos** [comment]
Not fixed in iOS 12.1.

**Comment by defagos** [comment]
Still not fixed in iOS 13.5.

**Comment by defagos** [comment]
Fixed as of iOS 14. I closed my radar accordingly.
